src/foremast/utils/gate.py

```diff
@@ -0,0 +1,50 @@
+#   Foremast - Pipeline Tooling
+#
+#   Copyright 2019 Redbox Automated Retail, LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Centralized Methods interacting with the Spinnaker Gate API."""
+import logging
+import requests
+
+from ..consts import API_URL, GATE_CA_BUNDLE, GATE_CLIENT_CERT
+
+LOG = logging.getLogger(__name__)
+OAUTH_ENABLED = False
+
+
+def gate_request(method='GET', uri=None, headers=None, data=None, params=None):
+    """Make a request to Gate's API via various auth methods
+
+    Args:
+        method (str): Method to request Gate API; GET or POST
+        uri (str): URI path to gate API
+    """
+    response = None
```

Comment:
You should do a `method = method.lower() #or method.upper()` to prevent silly issues like `gate_request(method='Post'...)` not working.

Reply:
Good idea!